nextroute/schema/input.go

```diff
@@ -7,129 +7,217 @@ import (
+	// Options arbitrary options.
+	Options any `json:"options,omitempty"`
+	// CustomData arbitrary custom data.
+	CustomData any `json:"custom_data,omitempty"`
+	// Defaults default properties for vehicles and stops.
+	Defaults *Defaults `json:"defaults,omitempty"`
+	// StopGroups group of stops that must be part of the same route.
+	StopGroups *[][]string `json:"stop_groups,omitempty"`
+	// DurationMatrix matrix of durations in seconds between stops.
+	DurationMatrix *[][]float64 `json:"duration_matrix,omitempty"`
+	// DistanceMatrix matrix of distances in meters between stops.
+	DistanceMatrix *[][]float64 `json:"distance_matrix,omitempty"`
+	// DurationGroups duration in seconds added when approaching the group.
 	DurationGroups *[]DurationGroup `json:"duration_groups,omitempty"`
-	Vehicles       []Vehicle        `json:"vehicles,omitempty"`
-	Stops          []Stop           `json:"stops,omitempty"`
+	// Vehicles to route.
+	Vehicles []Vehicle `json:"vehicles,omitempty"`
+	// Stops that will be routed and assigned to the vehicles.
+	Stops []Stop `json:"stops,omitempty"`
+	// AlternateStops a set of alternate stops for vehicles.
 	AlternateStops *[]AlternateStop `json:"alternate_stops,omitempty"`
 }
 
 // Defaults contains default values for vehicles and stops.
 type Defaults struct {
+	// Vehicles default values for vehicles.
 	Vehicles *VehicleDefaults `json:"vehicles,omitempty"`
-	Stops    *StopDefaults    `json:"stops,omitempty"`
+	// Stops default values for stops.
+	Stops *StopDefaults `json:"stops,omitempty"`
 }
 
 // VehicleDefaults contains default values for vehicles.
 type VehicleDefaults struct {
-	Capacity                any        `json:"capacity,omitempty"`
-	StartLevel              any        `json:"start_level,omitempty"`
-	StartLocation           *Location  `json:"start_location,omitempty"`
-	EndLocation             *Location  `json:"end_location,omitempty"`
-	Speed                   *float64   `json:"speed,omitempty"`
-	StartTime               *time.Time `json:"start_time,omitempty"`
-	EndTime                 *time.Time `json:"end_time,omitempty"`
-	MinStops                *int       `json:"min_stops,omitempty"`
-	MinStopsPenalty         *float64   `json:"min_stops_penalty,omitempty"`
-	MaxStops                *int       `json:"max_stops,omitempty"`
-	MaxDistance             *int       `json:"max_distance,omitempty"`
-	MaxDuration             *int       `json:"max_duration,omitempty"`
-	MaxWait                 *int       `json:"max_wait,omitempty"`
-	CompatibilityAttributes *[]string  `json:"compatibility_attributes,omitempty"`
-	ActivationPenalty       *int       `json:"activation_penalty,omitempty"`
-	AlternateStops          *[]string  `json:"alternate_stops,omitempty"`
+	// Capacity of the vehicle.
+	Capacity any `json:"capacity,omitempty"`
+	// StartLevel initial level of the vehicle.
+	StartLevel any `json:"start_level,omitempty"`
+	// StartLocation location where the vehicle starts..
```

Comment:
```suggestion
	// StartLocation location where the vehicle starts.
```
I would add something like: "If omitted, the vehicle starts at the first stop it services."